### PR TITLE
More fixes to searching in shares

### LINKED ISSUE #3700: Search works incorrect after MOVE request
ocis local: 
branch: v2.0.0-alpha1

Preparation: 
- rm -rf /.ocis
- run ocis
- admin creates New folder
- inside the "New folder" folder create: folder "test" and file "test.txt"

Steps: 
- in the home page try to search by "test" - work correct
Screenshot 2022-05-05 at 16 03 47

- try to send request "MOVE" (rename or remove to trash) file "test.txt" or folder "test"
- in the home page try again to search by "test" - 
Screenshot 2022-05-05 at 16 08 53

Actual: after renaming file, search finds it in "personal", but file is in a different location

### LINKED ISSUE #3701: Search is case-sensitive
ocis local:
branch: v2.0.0-alpha1

Steps:
- admin creates New folder
- inside the "New folder" folder create file "test.txt"
- in the home page try to search by "Test" 

Actual: search result is empty

### PR DESCRIPTION
**Pull request by aduffeck**
This PR fixes a problem when searching in sharey by making the search results relative to the mountpoint space instead of the grant space.
It also adds more debug logs and changes the search to still continue and return whatever it can find in case there's a problem searching within one space.

Fixes #3701
Fixes #3700

### CODE REVIEW

**Comment by ownclouders** [comment]
:boom: Acceptance test [<strong>Core-API-Tests-ocis-storage-4</strong>](https://drone.owncloud.com/owncloud/ocis/11538/44/1) failed. Further test are cancelled...

File: extensions/search/pkg/search/index/index.go [hunk 1]
```diff
@@ -232,10 +239,29 @@ func (i *Index) Search(ctx context.Context, req *searchsvc.SearchIndexRequest) (
 }
 
 // BuildMapping builds a bleve index mapping which can be used for indexing
-func BuildMapping() mapping.IndexMapping {
+func BuildMapping() (mapping.IndexMapping, error) {
+	NameMapping := bleve.NewTextFieldMapping()
+	NameMapping.Analyzer = "lowercaseKeyword"
```
Comment (kobergj) [inline extensions/search/pkg/search/index/index.go:244]: Maybe Lowercase naming for `nameMapping`? Or you have a special reason for uppercase?

File: extensions/search/pkg/search/provider/events.go [hunk 2]
```diff
@@ -126,3 +135,20 @@ func (p *Provider) statResource(ref *provider.Reference, owner *user.User) (*pro
...
+func (p *Provider) getPath(id *provider.ResourceId, owner *user.User) (*provider.GetPathResponse, error) {
+	// Get auth
+	ownerCtx := ctxpkg.ContextSetUser(context.Background(), owner)
+	authRes, err := p.gwClient.Authenticate(ownerCtx, &gateway.AuthenticateRequest{
+		Type:         "machine",
+		ClientId:     "userid:" + owner.Id.OpaqueId,
+		ClientSecret: p.machineAuthAPIKey,
+	})
+	if err != nil || authRes.GetStatus().GetCode() != rpc.Code_CODE_OK {
+		p.logger.Error().Err(err).Interface("authRes", authRes).Msg("error using machine auth")
```
Comment (kobergj) [inline extensions/search/pkg/search/provider/events.go:148]: We want to `return` here, right?

**Review by kobergj — CHANGES_REQUESTED** [review]
Just some open questions, rest looks :+1:

**Pull request merged**